cmd/worker.go

```diff
@@ -0,0 +1,18 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"github.com/tsaikd/gogstash/config/goglog"
+)
+
+func waitSignals(ctx context.Context) error {
+	osSignalChan := make(chan os.Signal, 1)
+	signal.Notify(osSignalChan, os.Interrupt, os.Kill)
+
+	sig := <-osSignalChan
```

Comment:
will app hang on this when ctx cancel ?

```
select {
case <- ctx.Done():
    return nil
case sig := <-osSignalChan:
    // blahblah...
```

Reply:
Oops, https://github.com/tsaikd/gogstash/pull/59/commits/e7314ff384035e4980250c22fd0a4231ae33c203 this will end with `cancel`, I think it will be ok?

Reply:
Sorry, I misunderstand, https://github.com/tsaikd/gogstash/pull/59/commits/18da6cd399e664387d5ae7c50d470719d80fecd6 this will be fine

Reply:
Oh! you are right, app will final do `cancel` this time https://github.com/tsaikd/gogstash/pull/59/commits/ee162c67dcdfc2c69ee4cb533a80ae023c3b8675